File: 2-semestre/uri-online-judge/c/1094.c
#include <stdio.h>
#include <string.h>

int main() {
    int n, i;
    char strAni[] = "X";
    
    scanf("%d", &n);
    
    
    int numCob = 0;
    int totCob = 0;
    char C[] = "C";
    char R[] = "R";
    char S[] = "S";
    int totCoe = 0;
    int totRat = 0;
    int totSap = 0;

    for (i = 0; i < n; i++) {
        scanf("%d", &numCob);
        totCob += numCob;

        scanf("%s", strAni);
        if (0 == strcmp(strAni, C)) totCoe += numCob;
        if (0 == strcmp(strAni, R)) totRat += numCob;
        if (0 == strcmp(strAni, S)) totSap += numCob;
    }

    double perCoe = (totCoe * 100) / (double)totCob;
    double perRat = (totRat * 100) / (double)totCob;
    double perSap = (totSap * 100) / (double)totCob;

    printf("Total: %d cobaias\n", totCob);
    printf("Total de coelhos: %d\n", totCoe);
    printf("Total de ratos: %d\n", totRat);
    printf("Total de sapos: %d\n", totSap);
    printf("Percentual de coelhos: %.2f %%\n", perCoe);
    printf("Percentual de ratos: %.2f %%\n", perRat);
    printf("Percentual de sapos: %.2f %%\n", perSap);
    
    return 0;
}

File: 2-semestre/uri-online-judge/java/1094.java
import java.io.IOException;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) throws IOException {
        Scanner entrada = new Scanner(System.in);
        
        int n = entrada.nextInt();
        int[] numCob = new int[n];
        String[] strAni = new String[n];
        
        for (int i = 0; i < n; i++) {
            numCob[i] = entrada.nextInt();
            strAni[i] = entrada.next();
        }
        
        int totCob = 0;
        int totCoe = 0;
        int totRat = 0;
        int totSap = 0;
        
        for (int i = 0; i < numCob.length; i++) {
            totCob += numCob[i];
        }
        
        for (int i = 0; i < n; i++) {
            if (strAni[i].equals("C")) {
                totCoe += numCob[i];
            } else if (strAni[i].equals("R")) {
                totRat += numCob[i];
            } else if (strAni[i].equals("S")) {
                totSap += numCob[i];
            }
        }

        double percCoe = (totCoe * 100) / (double)totCob;
        double percRat = (totRat * 100) / (double)totCob;
        double percSap = (totSap * 100) / (double)totCob;

        System.out.printf("Total: %d cobaias\n", totCob);
        System.out.printf("Total de coelhos: %d\n", totCoe);
        System.out.printf("Total de ratos: %d\n", totRat);
        System.out.printf("Total de sapos: %d\n", totSap);
        System.out.printf("Percentual de coelhos: %.2f %%\n", percCoe);
        System.out.printf("Percentual de ratos: %.2f %%\n", percRat);
        System.out.printf("Percentual de sapos: %.2f %%\n", percSap);
    }
}
